File: lky_msa/mstudy/src/main/java/com/study/mstudy/feign/HistoryFeignClient.java
package com.study.mstudy.feign;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.cloud.openfeign.FeignClient;
import java.util.Map;

@FeignClient(name = "history-service")
public interface HistoryFeignClient {

    @RequestMapping(method = RequestMethod.POST, value = "/v1/history/save")
    String saveHistory(Map<String, Object> paramMap);
}

File: k_rstudy/src/main/kotlin/com/krstudy/kapi/global/app/HistoryFeignClient.kt
package com.krstudy.kapi.global.app

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody

@FeignClient(name = "history-service")
interface HistoryFeignClient {

    @PostMapping("/v1/history/save")
    fun saveHistory(@RequestBody paramMap: Map<String, Any>): String
}